app/controllers/course/forum/forums_controller.rb

```diff
@@ -0,0 +1,85 @@
+class Course::Forum::ForumsController < Course::ComponentController
+  before_action :load_forum, except: [:index, :new, :create]
+  load_and_authorize_resource :forum, class: Course::Forum.name, through: :course,
+                                      only: [:index, :new, :create]
+
+  def index
+    @forums = @forums.with_topic_view_count.with_topic_count.with_topic_post_count.
+              includes(:subscriptions)
+  end
+
+  def show
+    @topics = @forum.topics.accessible_by(current_ability).order(created_at: :desc).
+              page(params[:page])
+  end
+
+  def new
+  end
+
+  def create
+    if @forum.save
+      redirect_to course_forums_path(current_course), success: t('.success', name: @forum.name)
+    else
+      render 'new'
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @forum.update(forum_params)
+      redirect_to course_forum_path(current_course, @forum),
+                  success: t('.success', name: @forum.name)
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    if @forum.destroy
+      redirect_to course_forums_path(current_course), success: t('.success', name: @forum.name)
+    else
+      redirect_to course_forums_path(current_course),
```

Comment:
I understand. So you mean I should also provide `delete` button in edit page right?